packages/client/src/index.ts

```diff
@@ -359,11 +362,27 @@ export function compileWorkflowOptions({
 }
 
 /**
- * Convert {@link ConnectionOptions.tls} to {@link grpc.ChannelCredentials} and
- * add the grpc.ssl_target_name_override GRPC {@link ConnectionOptions.channelArgs}
+ * Normalize {@link ConnectionOptions.tls} by turning false and null to undefined and true to and empty object
+ * NOTE: this function is duplicated in `packages/worker/src/worker.ts` for lack of a shared library
+ */
+function normalizeTlsConfig(tls?: ConnectionOptions['tls']): TLSConfig | undefined {
+  return typeof tls === 'object' ? (tls === null ? undefined : tls) : tls ? {} : undefined;
```

Comment:
This is so painfully javascript :laughing:

Reply:
Yup ..